src/main/java/net/fabricmc/loader/entrypoint/EntrypointTransformer.java

```diff
@@ -39,7 +38,7 @@
 	private boolean entrypointsLocated = false;
 
 	public EntrypointTransformer(Function<EntrypointTransformer, List<EntrypointPatch>> patches) {
-		this.patches = ImmutableList.copyOf(patches.apply(this));
+		this.patches = patches.apply(this);
```

Comment:
Could maybe use `Collections.unmodifiableList` to keep the same guarantees ?

Reply:
I don't see a point behind the copy or immutability, this should be all internal without anything encouraging modification.